### ISSUE TITLE
Display login failure messages

### ISSUE THREAD

**Issue by nils-werner** [drone/drone#660]
When using Drone with GitLab and using invalid credentials, the page shows a white screen.
It would be helpful for the end user if there was an error message shown.

**Comment by mattbostock** [comment]
Should this bug still be assigned to milestone v0.3.1 if #814 was closed? Will the new UI be implemented for v0.3.1?

**Comment by bscott** [comment]
I get this error even when I'm already signed into Github with correct creds

**Comment by bradrydzewski** [comment]
what do you see in the logs? Each potential error path in the authorization code is logged to help correctly identify the issue. There could be a number of problems. You could have closed registration enabled, org restrictions, incorrect redirect URL mismatch with your GitHub settings and more. The logs will indicate which issue is being encountered.

**Comment by brunowego** [comment]
+1

---
Log

```log
2015/07/29 19:55:31 Error exchanging token. OAuthError: updateToken: Unexpected HTTP status 404 Not Found
2015/07/29 19:55:39 Error matching state in OAuth2 redirect
```

**Comment by bradrydzewski** [comment]
this has been fixed in the 0.4 branch, so I'm closing
